### PR TITLE
Introducing the stargazing minigame

### PR DESCRIPTION
**Pull request by marinprusac**
The minigame is now mostly complete for the first prototype. It will be necessary to integrate it into the rest of the game.

### CODE REVIEW

File: stargazing/star_picker_logic.gd [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+signal calibrated(factor: float)
+signal solved_star()
+
+const focus_duration: float = 1
+
+var horizontal_angle = 0
+var vertical_angle = 0
+var calibration = 0
+
+var stars = []
```
Comment (savipauk) [inline stargazing/star_picker_logic.gd]: These are dynamically typed

File: stargazing/star_picker_logic.gd [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+func _process(delta: float) -> void:
+	if len(stars) == 0:
+		return
+	
+	var min_dist = 1
+	var chosen_star = stars[0]
+	
+	for star in stars:
```
Comment (savipauk) [inline stargazing/star_picker_logic.gd]: Dynamically typed

File: stargazing/stargazer_perspective.gd [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+@export var rotate_speed = 1/PI
+@export var min_vertical_angle = PI/8
+@export var max_vertical_angle = 3*PI/8
+@export var instability = 0.1
+
+var horizontal_angle = 0
+var vertical_angle = PI/2
+
+var instability_dir = Vector2.RIGHT
+var instability_elapsed = 0
```
Comment (savipauk) [inline stargazing/stargazer_perspective.gd]: Dynamically typed

File: stargazing/stargazing_tutorial.gd [hunk 4]
```diff
@@ -0,0 +1,26 @@
+extends Node
+
+signal load_star_group(star_group_index: int)
+signal set_completion(group: int, star: int)
+signal all_solved()
+
+var star_groups = [StarAngles.cassiopeia_angles, StarAngles.little_dipper_angles, StarAngles.big_dipper_angles, StarAngles.little_dipper_angles[6]]
+var solved_groups = 0
+var solved_stars = 0
```
Comment (savipauk) [inline stargazing/stargazing_tutorial.gd]: Dynamically typed

**Review by savipauk — CHANGES_REQUESTED** [review]
looks good, some stuff is dynamically typed

File: stargazing/star_angles.gd [hunk 5]
```diff
@@ -0,0 +1,41 @@
+extends Node
+class_name StarAngles
+
+static var big_dipper_angles = [
```
Comment (savipauk) [inline stargazing/star_angles.gd]: this can also be `big_dipper_angles: Array[Vector2]`

**Review by savipauk — CHANGES_REQUESTED** [review]

**Review by dean-rakic — APPROVED** [review]

**Comment by marinprusac** [comment]
Dynamic typing removed, re-requesting review

**Review by savipauk — APPROVED** [review]

**Review by JelenaGlasovac — APPROVED** [review]
Looks good!

**Pull request merged**